bciers/apps/reporting/src/app/components/operations/OperationReview.tsx

```diff
@@ -1,144 +1,197 @@
-    body: JSON.stringify(formDataObject),
-    headers: {
-      "Content-Type": "application/json",
-      accept: "application/json",
-    },
-  });
+const baseUrl = "/reports";
+const cancelUrl = "/reports";
+
+const taskListElements: TaskListElement[] = [
+  {
+    type: "Section",
+    title: "Operation information",
+    isExpanded: true,
+    elements: [
+      { type: "Page", title: "Review Operation information", isActive: true },
+      { type: "Page", title: "Person responsible" },
+      { type: "Page", title: "Review facilities" },
+    ],
+  },
+];
+
+const formatDate = (dateString: string | number | Date) => {
+  return dayjs(dateString).format("MMM DD YYYY");
 };
 
 export default function OperationReview({
   formData,
   version_id,
-}: Readonly<Props>) {
-  const customStepNames = [
-    "Operation Information",
-    "Facilities Information",
-    "Compliance Summary",
-    "Sign-off & Submit",
-  ];
+  reportingYear,
+  allActivities,
+  allRegulatedProducts,
+}: Props) {
+  const router = useRouter();
+  const [schema, setSchema] = useState<RJSFSchema>(operationReviewSchema);
+  const [uiSchema] = useState<any>(operationReviewUiSchema);
+  const [formDataState, setFormDataState] = useState<any>(formData);
+  const [loading, setLoading] = useState(true);
+  const saveAndContinueUrl = `/reporting/reports/${version_id}/person-responsible`;
+  const reportingWindowEnd = reportingYear?.reporting_window_end
+    ? formatDate(reportingYear.reporting_window_end)
+    : null;
+
+  const submitHandler = async (
+    data: { formData?: any },
+    reportVersionId: number,
+  ) => {
+    const method = "POST";
+    const endpoint = `reporting/report-version/${reportVersionId}/report-operation`;
+    const pathToRevalidate = `reporting/report-version/${reportVersionId}/report-operation`;
+
+    // Extract the formData from the input
+    const formDataObject = safeJsonParse(JSON.stringify(data.formData));
+
+    // Ensure activities and regulated_products are present and correctly formatted
+    const preparedData = {
+      ...formDataObject,
+      activities: formDataObject.activities?.map((activityId: any) => {
+        const activityName = allActivities.find((a) => a.id === activityId)
+          ?.name;
+        return activityName || activityId;
+      }),
+      regulated_products: formDataObject.regulated_products?.map(
+        (productId: any) => {
+          const productName = allRegulatedProducts.find(
+            (p) => p.id === productId,
+          )?.name;
+          return productName || productId;
+        },
+      ),
+    };
+
+    try {
+      await actionHandler(endpoint, method, pathToRevalidate, {
+        body: JSON.stringify(preparedData),
+        headers: {
+          "Content-Type": "application/json",
+          accept: "application/json",
+        },
+      });
+      router.push(saveAndContinueUrl);
+    } catch (error) {
+      console.error("Submission failed:", error);
+    }
```

Comment:
Unnecessary error handling, the `actionHandler` already has its own error handling (Sentry, and some logging), and will never throw.

`actionHandler` also returns a `response` object of type [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) if you want to check the status: `response.ok`, `response.status` are much better bets.